src/protocol/IDataFeed.sol

```diff
@@ -2,44 +2,25 @@
 pragma solidity ^0.8.28;
 
 interface IDataFeed {
-    struct HookQuery {
-        address provider;
-        bytes input;
-        uint256 value;
-    }
-
-    struct Publication {
+    struct PublicationHeader {
         uint256 id;
         bytes32 prevHash;
         address publisher;
         uint256 timestamp;
         uint256 blockNumber;
```

Comment:
We can pack publisher, blockNumber and timestamp within a single slot (i.e. (uint160, uint48, uint48)). Not to do it now as the design may change again but to consider later.